pkg/interactive/inputs/request_loader/loader.go

```diff
@@ -0,0 +1,108 @@
+package requestloader
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/apono-io/apono-cli/pkg/aponoapi"
+	"github.com/apono-io/apono-cli/pkg/clientapi"
+	"github.com/apono-io/apono-cli/pkg/services"
+
+	"github.com/charmbracelet/bubbles/spinner"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/gookit/color"
+)
+
+const (
+	abortingText = "Aborting..."
+)
+
+func (m model) Init() tea.Cmd {
+	return tea.Batch(waitForRequest(m.ctx, m.client, m.creationTime, m.timeout), m.spinner.Tick)
+}
+
+func (e errMsg) Error() string { return e.err.Error() }
+
+func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case statusMsg:
+		m.request = (*clientapi.AccessRequestClientModel)(&msg)
+		if m.noWaitForGrant || shouldStopLoading(m.request) {
+			m.quitting = true
+			return m, tea.Quit
+		}
+		if time.Now().After(m.creationTime.Add(m.timeout)) {
+			m.err = fmt.Errorf("timeout waiting for request to be granted")
+			return m, tea.Quit
+		}
+
+		return m, getUpdatedRequest(m.ctx, m.client, m.request.Id)
+
+	case errMsg:
+		m.err = msg
+		return m, tea.Quit
+
+	case tea.KeyMsg:
+		if msg.String() == abortKey {
+			m.aborting = true
+			return m, tea.Quit
+		}
+	case spinner.TickMsg:
+		var cmd tea.Cmd
+		m.spinner, cmd = m.spinner.Update(msg)
+		return m, cmd
+	}
+
+	return m, nil
+}
+
+func (m model) View() string {
+	if m.quitting {
+		return ""
+	}
+	if m.aborting {
+		return abortingText
+	}
+
+	var msg string
+	if m.request == nil {
+		msg = fmt.Sprintf("%s Waiting for request to be created", m.spinner.View())
+	} else {
+		msg = fmt.Sprintf("%s Request %s is %s", m.spinner.View(), color.Bold.Sprint(m.request.Id), services.ColoredStatus(*m.request))
+	}
+
+	return "\n" + msg + "\n\n"
```

Comment:
new lines are needed?

Reply:
for better output yes, its only when the loader is present